### PR TITLE
doc/moved environment variable warning before cmake

### PR DESCRIPTION
**Pull request by cbkerr**
## Description
Following the build directions rendered on github, I skipped setting the `CMAKE_PREFIX_PATH` because the warning appeared after the step at which it is needed. (I thought it only applied to the `make` step.) Move warning one paragraph earlier.

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/master/ContributorAgreement.md).
- [x] My name is on the list of contributors (`sphinx-doc/credits.rst`) in the pull request source branch.

### CODE REVIEW

File: INSTALLING.rst [hunk 1]
```diff
@@ -180,19 +180,19 @@ Run ``python3 install-prereq-headers.py -h`` to see a list of the command line o
 Compile HOOMD-blue
 ------------------
 
-Configure::
-
-    $ cd /path/to/hoomd-blue
-    $ cmake -B build
-    $ cd build
-
 .. warning::
```
Comment (joaander) [inline INSTALLING.rst:183]: The section *Configure a virtual environment* includes instructions to set `CMAKE_PREFIX_PATH`. If that is too far from the `cmake` command that requires, I think we should move that here and delete this warning block.

**Review by joaander — CHANGES_REQUESTED** [review]
Thanks. I have a small suggestion to make things more clear.

File: INSTALLING.rst (hunk 1 above)
Comment (cbkerr) [reply INSTALLING.rst]: Thanks for pointing that out!

**Comment by cbkerr** [comment]
I moved the section about virtual environments after the section on install prerequisites, allowing us to delete the warning. Instead, I added a normal text reminder about the environment variable.

While reading for flow, I moved another bit of text above the corresponding command.

I also built it locally to make sure it builds.

**Comment by joaander** [comment]
Do you find it a problem that now *Install prerequisites* mentions installing into a virtual environment that doesn't exist yet?

**Comment by cbkerr** [comment]
> Do you find it a problem that now _Install prerequisites_ mentions installing into a virtual environment that doesn't exist yet?

Good point. I was treating "install prerequisites" as "the install prerequisites" and it felt better to have the "prerequisites" first. A virtual environment is kind of a prerequisite, but it is different from a system package manager or module system. I think it makes sense to discuss the general case before talking about virtual environments.

I've tried restructuring some of the end of the "dependencies" section and merged it with the virtual environment section. It helped to split discussion of the module system from the discussion of virtual environments.

**Review by joaander — APPROVED** [review]
Thanks!

**Pull request merged**